openllm-python/src/openllm/protocol/openai.py

```diff
@@ -0,0 +1,43 @@
+from __future__ import annotations
+import time
+
+import attr
+
+import openllm_core
+
+@attr.define
+class LogProbs:
+    text_offset: list[int] = attr.ib(default=attr.Factory(list))
```

Comment:
Let's use `attr.field` instead of ib.

Reply:
Note that since we are compat with 3.8, the type hint will fail.

The current type annotations that you have is according to [PEP604](https://peps.python.org/pep-0604/), which we are currently using. But for attrs class that does serialisation. Best to do just do 

```python

import typing as t

@attr.define
class LogProbs:
  text_offset: t.List[int] = ...
```